CheckCompileLinkOption fails when Xcode compiler is invoked from command line

When testing for the compiler capabilities with:

    CheckCompileLinkOption("-msse2" CRYPTOPP_IA32_SSE2 "${TEST_PROG_DIR}/test_x86_sse2.cxx")

`CheckCompileLinkOption` invokes the `CMAKE_CXX_COMPILER` directly and compiles the test file. This file will fail to compile with an error message that it can't locate `stdlib.h`, when the compiler is clang from the xcode toolchain. Try to run:

    /Applications/Xcode9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -msse2 test_x86_sse2.cxx

This is the error:
```
In file included from test_x86_sse2.cxx:1:
In file included from /Applications/Xcode9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.0.0/include/emmintrin.h:27:
In file included from /Applications/Xcode9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.0.0/include/xmmintrin.h:39:
In file included from /Applications/Xcode9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/9.0.0/include/mm_malloc.h:27:
/Applications/Xcode9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/stdlib.h:94:15: fatal error: 'stdlib.h' file not found
#include_next <stdlib.h>
              ^~~~~~~~~~
1 error generated.
```

However as soon as you add the `-isysroot` pointing to the correct sysroot, compiles without a problem.

This seems to be a problem with the compilers in this path in the toolchain, the clang in `/usr/bin/` works fine.

So, there should be a way to modify `CmakeLists` to take the sysroot into account, but I'm not sure what would be the elegant way to do that...

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -225,6 +225,18 @@ function(CheckCompileLinkOption opt var prog)
 
     # TODO: improve this...
     CHECK_CXX_COMPILER_FLAG(${opt} ${var})
+    
+  elseif (APPLE)
+
+    message(STATUS "Performing Test ${var}")
+    try_compile(COMMAND_SUCCESS ${CMAKE_BINARY_DIR} ${prog} COMPILE_DEFINITIONS ${opt})
+    if (COMMAND_SUCCESS)
+      set(${var} 1 PARENT_SCOPE)
+      message(STATUS "Performing Test ${var} - Success")
+    else ()
+      set(${var} 0 PARENT_SCOPE)
+      message(STATUS "Performing Test ${var} - Failed")
+    endif ()
 
   else ()
 

